src/main/java/mate/academy/service/impl/PaymentServiceImpl.java

```diff
@@ -0,0 +1,212 @@
+package mate.academy.service.impl;
+
+import com.stripe.Stripe;
+import com.stripe.exception.StripeException;
+import com.stripe.model.checkout.Session;
+import com.stripe.param.checkout.SessionCreateParams;
+import jakarta.annotation.PostConstruct;
+import java.math.BigDecimal;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.payment.PaymentCancelResponseDto;
+import mate.academy.dto.payment.PaymentDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.mapper.PaymentMapper;
+import mate.academy.model.Booking;
+import mate.academy.model.Payment;
+import mate.academy.repository.PaymentRepository;
+import mate.academy.repository.booking.BookingRepository;
+import mate.academy.service.PaymentService;
+import mate.academy.service.StripeCheckoutService;
+import mate.academy.service.TelegramNotificationService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class PaymentServiceImpl implements PaymentService {
+
+    private final PaymentMapper paymentMapper;
+    private final PaymentRepository paymentRepository;
+    private final BookingRepository bookingRepository;
+    private final StripeCheckoutService stripeCheckoutService;
+    private final TelegramNotificationService telegramNotificationService;
+
+    @Value("${STRIPE_SECRET_KEY}")
+    private String stripeSecretKey;
+
+    @PostConstruct
+    public void init() {
+        Stripe.apiKey = stripeSecretKey;
+    }
+
+    @Override
+    @Transactional
+    public Payment initiatePayment(Long bookingId,
+                                   String sessionUrl,
+                                   BigDecimal amountToPay) {
+        Booking booking = bookingRepository
+                .findById(bookingId).orElseThrow(() ->
+                        new RuntimeException("Booking not found"));
```

Comment:
provide parameter to message
```suggestion
                        new EntityNotFoundException("Booking not found"));
```